packages/workbox-strategies/src/StrategyHandler.ts

```diff
@@ -556,13 +556,17 @@ class StrategyHandler {
    *
    * Note: any work done after `doneWaiting()` settles should be manually
    * passed to an event's `waitUntil()` method (not this handler's
-   * `waitUntil()` method), otherwise the service worker thread my be killed
+   * `waitUntil()` method), otherwise the service worker thread may be killed
    * prior to your work completing.
    */
   async doneWaiting(): Promise<void> {
-    let promise;
-    while ((promise = this._extendLifetimePromises.shift())) {
-      await promise;
+    while (this._extendLifetimePromises.length) {
```

Comment:
I suppose the `while` loop needs to be retained in case additional promises are added to the array while awaiting the ones that were removed during the first iteration?

Reply:
Yes. (That was my rationale, at least.)